lib/ui/painting.dart

```diff
@@ -4935,26 +4591,1632 @@ class PictureRecorder extends NativeFieldWrapperClass2 {
+  }
+
+  @override
+  void save() {
+    _addOp(_CanvasOp.save, 0);
+    _ctmStack.add(_MatrixTransform._copy(_ctm));
+  }
+
+  @override
+  void saveLayer(Rect? bounds, Paint paint) {
+    _updatePaintData(paint, _saveLayerMask);
+    if (bounds == null) {
+      _addOp(_CanvasOp.saveLayer, 0);
+    } else {
+      _addOp(_CanvasOp.saveLayerBounds, _nRectData);
+      _addRect(bounds);
+    }
+    _ctmStack.add(_MatrixTransform._copy(_ctm));
+  }
+
+  @override
+  int getSaveCount() => _ctmStack.length;
+
+  @override
+  void restore() {
+    if (_ctmStack.isNotEmpty) {
+      _ctm = _ctmStack.removeLast();
+      _addOp(_CanvasOp.restore, 0);
+    }
+  }
+  @override
+  void translate(double dx, double dy) {
+    _addOp(_CanvasOp.translate, 2);
+    _addScalar2(dx, dy);
+    _ctm.translate(dx, dy);
+  }
+
+  @override
+  void scale(double sx, [ double? sy ]) {
+    _addOp(_CanvasOp.scale, 2);
+    _addScalar2(sx, sy ?? sx);
+    _ctm.scale(sx, sy ?? sx);
+  }
+
+  @override
+  void rotate(double radians) {
+    _addOp(_CanvasOp.rotate, 1);
+    _addScalar(radians);
+    _ctm.rotate(radians);
+  }
+
+  @override
+  void skew(double sx, double sy) {
+    _addOp(_CanvasOp.skew, 2);
+    _addScalar2(sx, sy);
+    _ctm.skew(sx, sy);
+  }
+
+  @override
+  void transform(Float64List matrix4) {
+    if (matrix4.length != 16)
+      throw ArgumentError('"matrix4" must have 16 entries.');
+    if (matrix4[3] == 0.0 && matrix4[7] == 0.0 && matrix4[15] == 1.0) {
+      _addOp(_CanvasOp.transform2x3, 6);
+      _addScalar3(matrix4[0], matrix4[4], matrix4[12]);
+      _addScalar3(matrix4[1], matrix4[5], matrix4[13]);
+    } else {
+      _addOp(_CanvasOp.transform3x3, 9);
+      _addScalar3(matrix4[0], matrix4[4], matrix4[12]);
+      _addScalar3(matrix4[1], matrix4[5], matrix4[13]);
+      _addScalar3(matrix4[3], matrix4[7], matrix4[15]);
+    }
+    _ctm.mul(matrix4[0], matrix4[4], matrix4[12],
+             matrix4[1], matrix4[5], matrix4[13]);
+  }
+
+  @override
+  void clipRect(Rect rect, {ClipOp clipOp = ClipOp.intersect, bool doAntiAlias = false}) {
+    switch (clipOp) {
+      case ClipOp.intersect:
+        _addOp(doAntiAlias ? _CanvasOp.clipRectAA : _CanvasOp.clipRect, _nRectData);
+        break;
+      case ClipOp.difference:
+        _addOp(doAntiAlias ? _CanvasOp.clipRectAADiff : _CanvasOp.clipRectDiff, _nRectData);
+        break;
+    }
+    _addRect(rect);
+  }
+
+  @override
+  void clipRRect(RRect rrect, {bool doAntiAlias = false}) {
+    if (rrect.isRect) {
+      clipRect(rrect.outerRect, doAntiAlias: doAntiAlias);
+    } else {
+      _addOp(doAntiAlias ? _CanvasOp.clipRRectAA : _CanvasOp.clipRRect, _nRoundRectData);
+      _addRRect(rrect);
+    }
+  }
+
+  @override
+  void clipPath(Path path, {bool doAntiAlias = false}) {
+    _addOp(doAntiAlias ? _CanvasOp.clipPathAA : _CanvasOp.clipPath, 0);
+    _addPathData(path);
+  }
+
+  // These fields track the SkPaint defaults in SkPaint::SkPaint()
```

Comment:
If we expect to make this be true going forward, we might be in for a challenge since we don't own those defaults.

If not, we probably don't need this comment.